### PR TITLE
Remove Basic metadata

### PR DESCRIPTION
**Pull request by laritakr**
Fixes https://github.com/scientist-softserv/utk-hyku/issues/131:
- Modifies installers to handle removing BasicMetadata and IndexesBasicMetadata
- Adds FoundationalMetadata to include only those attributes which cannot be defined via the metadata modeling yaml file

Other fixes:
- Modifies install generator to handle adding menu option if needed (Bulkrax adds the menu to the app, so menu here was not being reached)
- Modifies work generator to correct https://github.com/samvera-labs/allinson_flex/pull/102

### CODE REVIEW

File: app/models/allinson_flex/profile_class.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ module AllinsonFlex
   class ProfileClass < ApplicationRecord
     # @todo any before_destroy validation?
     # many to many relationship between class and context
-    has_and_belongs_to_many :profile_contexts
+    has_and_belongs_to_many :profile_contexts # this seems incorrect??
```
Comment (laritakr) [inline app/models/allinson_flex/profile_class.rb:7]: Added this note, as has_and_belongs_to_many is not defined on both models, and additionally there is a defined table, ProfileClassContext as the intersection of these two classes.

File: app/models/allinson_flex/profile_class_context.rb [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 module AllinsonFlex
   class ProfileClassContext < ApplicationRecord
+    self.table_name = 'allinson_flex_profile_classes_contexts'
```
Comment (laritakr) [inline app/models/allinson_flex/profile_class_context.rb:5]: Added due to errors thrown when trying to access this table via the rails console.

File: app/models/allinson_flex/profile_class.rb (hunk 1 above)
Comment (ShanaLMoore) [reply app/models/allinson_flex/profile_class.rb:7]: Could you update the 'this seems incorrect??' comment with the details provided in your PR comment? Having the extra context will be useful to us when we evaluate this later.

**Review by ShanaLMoore — APPROVED** [review]
nice job @laritakr 🎉

**Pull request merged**